src/dao/TapDAO.js

```diff
@@ -27,11 +28,38 @@ class TapDAO extends DAO<Tap, TapMutator> {
     });
   }
 
+  countLeaderboard(
+    tapID: EntityID,
+    duration: string,
+    queryOptions?: QueryOptions,
+  ): LoadObject<number> {
+    const funcString = `Default.leaderboard(timeSpan=duration'${duration}')`;
+    const stringifiedID = tapID.toString();
+
+    return this.__countCustom(
+      (countQueryOptions: QueryOptions): oHandler<Object> => {
+        const handler = this.__buildHandler(
+          {
+            ...queryOptions,
+            ...countQueryOptions,
+          },
+          false,
+        ).find(this.__reformatIDValue(stringifiedID));
+        handler.func(funcString);
+
+        return handler;
+      },
+      queryOptions,
+      funcString,
+    );
+  }
+
   fetchLeaderboard(
     tapID: EntityID,
+    duration: string,
     queryOptions?: QueryOptions,
   ): LoadObject<Array<LeaderboardItem>> {
-    const funcString = 'Default.leaderboard()';
+    const funcString = `Default.leaderboard(timeSpan=duration'${duration}')`;
```

Comment:
Don't you need to do this in the `KegsDAO` as well? or have you not added that code yet?

Reply:
haven't added. But don't need at the moment.  In the current app structure, we have leaderboard only on tap details page.

Reply:
Ok, in the old app you could see the leaderboard on the tap and current keg. I'm not sure if we want to do that again.